src/index.js

```diff
@@ -6,7 +6,18 @@ import {OpenForm} from './sdk';
 const {REACT_APP_BASE_API_URL, REACT_APP_FORM_ID, REACT_APP_USE_HASH_ROUTING} = process.env;
 
 window.onload = () => {
```

Comment:
index.js is not used in the production bundle, it is only relevant for the local development. The actual entry point for production bundles is `sdk.js`, and that's where all this stuff should be handled.

I think it should also be possible to translate the real query string param into a react-router (query) param on the root route, and have the redirect handled there. We do something like that already for appointment forms: https://github.com/open-formulieren/open-forms-sdk/blob/main/src/components/App.js#L74

Reply:
Ok now it makes sense why the `CSPNonce` was hardcoded. And yes my goal here is to avoid handling things manually and let react router do the redirect

Reply:
I've spent some time experimenting and it is a real pain. With https://github.com/open-formulieren/open-forms/pull/3574, the backend is currently doing redirects to the following `Location`: `http://localhost:3000/?_action=afspraak-annuleren&time=...&submission_uuid=...`.

The issue is that as soon as the router is "initialized", hash based routing will not take into account anything before `#` (in this case it is implicit and added at the end of the URL, meaning nothing is accessible through the router API, e.g. `useLocation().search == ''`).

I can use `document.location.search` instead, and return `Navigate` after applying `_action` logic, but I end up with the following URL:

`http://localhost:3000/?_action=afspraak-annuleren&time=...&submission_uuid=...#/afspraak-annuleren?time=...&submission_uuid=...`
because hash based routing knows nothing before `#` and consider it to be the basepath.

So I could play manually with `history.pushState` but this is dirty and doesn't play well with `react-router`. Any suggestions welcomed.

Reply:
What am I missing if I'm thinking of doing these steps?

1. In `src/components/App.js` get the global query params using `new URLSearchParams(window.location.search)`
2. Check if the "action" param is present. If there is, return the `<Navigate>` component to redirect
    1. Create a new (client side route) params struct by keeping all query params, dropping the `action` one
    2. Map the action to the client side route and use that for the `<Navigate>`

I'm expecting both hash based and normal URL based routes to present the same query string:

* `localhost:3000/subpath?action=resume&submission_id=123`
* `localhost:3000/subpath/?action=resume&submission_id=123#!/to-strip-of-in-the-backend`